bynder_sdk/client/upload_client.py

```diff
@@ -1,144 +1,129 @@
+        :param file_path: The path of the asset to be uploaded.
+        :param file_id: The uuid4 used to identify the file to be uploaded.
+        :return:
+            - chunks_count: The number of chunks in which the file is to
+        be uploaded.
+            - file_size: The size of the file to be uploaded.
         """
+        chunks_count = 0
+        file_size = 0
+        with open(file_path, "rb") as f:
+            file_size = os.stat(f.fileno()).st_size
+            chunks_count = math.ceil(
+                file_size / MAX_CHUNK_SIZE)
+            for chunk_nr, chunk in enumerate(_read_in_chunks(f)):
+                self._upload_chunk(file_id, chunk, chunk_nr)
+
+        return chunks_count, file_size
+
+    def _upload_chunk(self, file_id, data, chunk_nr):
+        self.session.headers['content-sha256'] = sha256(data).hexdigest()
         self.session.post(
-            '/v4/upload/',
-            data={
-                'id': init_data['s3file']['uploadid'],
-                'targetid': init_data['s3file']['targetid'],
-                'filename': init_data['s3_filename'],
-                'chunkNumber': part_nr
-            }
+            '/v7/file_cmds/upload/{}/chunk/{}'.format(file_id, chunk_nr),
+            data=data
         )
 
-    def _finalise_file(self, init_data, total_parts):
-        """ Finalises a completely uploaded file.
+    def _finalise_file(self, file_id, file_name, file_size, chunks_count):
+        """Finalises a completely uploaded file.
+        :param file_id: The file_id returned from the prepare endpoint.
+        :param file_name: The name of the file to be uploaded.
+        :param file_size: The size of the file to be uploaded.
+        :param chunks_count: The number of chunks in which the file was
+        uploaded.
+        :return: The correlation_id of save media request.
         """
-        return self.session.post(
-            '/v4/upload/{0}/'.format(init_data['s3file']['uploadid']),
+        response = self.session.post(
+            '/v7/file_cmds/upload/{}/finalise_api'.format(file_id),
             data={
-                'id': init_data['s3file']['uploadid'],
-                'targetid': init_data['s3file']['targetid'],
-                's3_filename': init_data['s3_filename'],
-                'chunks': total_parts
+                'fileName': file_name,
+                'fileSize': file_size,
+                'chunksCount': chunks_count,
+                "sha256": self.file_sha256,
+                "intent": "upload_main_uploader_asset",
             }
         )
 
-    def _save_media(self, import_id, data, media_id=None):
-        """ Saves a new media asset in Bynder. If media id is specified
-        in the query a new version of the asset will be saved. Otherwise
-        a new asset will be saved.
+        correlation_id = response.headers['x-api-correlation-id']
+        return correlation_id
+
+    def _save_media(self, file_id, data, media_id=None):
```

Comment:
Would make more sense maybe to split this up in two methods, since they're completely different?

And also good to check if the brandId is present in the data maybe, since it's a required arg in the body.

Reply:
I agree there can be a check for the brandId but they can be 2 methods or even 1 method since the functionality of the method is just to save the media and change the url based on a param personally I would prefer to have it as 1 method.

Reply:
Yeah I thought two methods would make the code a bit simpler, since with one method it would basically be one if/else with the two requests that have different args, except for the file_id.